pkg/daemon/update.go

```diff
@@ -325,6 +336,40 @@ func ExtractOSImage(imgURL string) (osImageContentDir string, err error) {
 	return
 }
 
+// ExtractOSImage extracts OS image content in a temporary directory under /run/machine-os-content/
+// and returns the path on successful extraction.
+// Note that since we do this in the MCD container, cluster proxy configuration must also be injected
+// into the container. See the MCD daemonset.
+func ExtractExtensionsImage(imgURL string) (osExtensionsImageContentDir string, err error) {
+	var registryConfig []string
+	if _, err := os.Stat(kubeletAuthFile); err == nil {
+		registryConfig = append(registryConfig, "--registry-config", kubeletAuthFile)
+	}
+	if err = os.MkdirAll(osExtensionsContentBaseDir, 0o755); err != nil {
+		err = fmt.Errorf("error creating directory %s: %w", osExtensionsContentBaseDir, err)
+		return
+	}
+
+	if osExtensionsImageContentDir, err = ioutil.TempDir(osExtensionsContentBaseDir, "os-extensions-content-"); err != nil {
+		return
+	}
+
+	// Extract the image
+	args := []string{"image", "extract", "--path", "/:" + osExtensionsImageContentDir}
+	args = append(args, registryConfig...)
+	args = append(args, imgURL)
+	if _, err = pivotutils.RunExtBackground(cmdRetriesCount, "oc", args...); err != nil {
+		// Workaround fixes for the environment where oc image extract fails.
+		// See https://bugzilla.redhat.com/show_bug.cgi?id=1862979
+		glog.Infof("Falling back to using podman cp to fetch OS image content")
```

Comment:
since extensions image is relatively small image, we can skip using oc and use podmanCopy directly too